### DISCUSSION TITLE
workspace capabilities in neovim

### DISCUSSION THREAD

**Discussion by AdamBD** [community/community#82191] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Are there plans to add a general workspace level prompt experience for the neovim plugin?

**Comment by tpope** [comment, marked as answer]
Support for this was recently added. It's documented under `:help copilot_workspace_folders`

```vimhelp
g:copilot_workspace_folders
                        A list of "workspace folders" or project roots that
                        Copilot may use to improve to improve the quality of
                        suggestions.
>
                        let g:copilot_workspace_folders =
                               \ ["~/Projects/myproject"]
<
                        You can also set b:workspace_folder for an individual
                        buffer and newly seen values will be added
                        automatically.
```

**Reply by ethanIncentify** [reply]
How can I tell whether copilot is aware of the workspace?

**Reply by tpope** [reply]
> How can I tell whether copilot is aware of the workspace?

There's not an official way to do that. But this should work for now at least:

```vim
:echo keys(copilot#Client().workspaceFolders)
```

**Edit**: updated to new function name

**Reply by damien-robotsix** [reply]
Hello I just tried the suggestion to check the workspace, it is probably a bit old but is doesn't work.

Get unknown function copilot#Agent

**Reply by tpope** [reply]
```vim
:echo keys(copilot#Client().workspaceFolders)
```

**Reply by damien-robotsix** [reply]
Thank you very much :)
